Implements warps menu

MittenzHugg opened the pull request.

MittenzHugg committed:
PAL region support, better U1.0 and J hooks.

MittenzHugg committed:
-add PAL hooks

MittenzHugg committed:
corrected all region hooks

MittenzHugg committed:
stage 1 hook correction

MittenzHugg committed:
wow I'm bad at this

MittenzHugg committed:
repair bad function calls

MittenzHugg committed:
implement warps menu

MittenzHugg committed:
add warp menu src code. lol

krimtonz reviewed (changes requested):
I tend to be a bit anal about formatting.

There is some spacing inconsistencies 

For example 

```C
menu_button_add(scene_menu,0,k+1,ent->name, activate_warp_proc, (void *)(data |((ent->index) & 0xFF)));
```

I'd prefer something along these lines
```C
menu_button_add(scene_menu, 0, k+1, ent->name, activate_warp_proc, (void *)(data | ((ent->index) & 0xFF)));
```

Event handlers are set up in a way that only events that are registered will call the function.  The `on_activate` parameter of `menu_button_add` will register the `MENU_EVENT_ACTIVATE` event only, so a check on what `event` is unnecessary.

MittenzHugg committed:
correct warp name capilazation
add ccw entrances
add ccw seasons sub category
split create wraps menu function into reusable subprocesses

MittenzHugg committed:
correct exit for CCW hub from grunty's lair
